### ISSUE TITLE
Unable to use npm module in typescript app

### ISSUE THREAD

**Issue by j3ddesign** [xjsframework/xjs#200]
I'm working on building an extension and write in typescript, however because the npm module is exporting typescript files instead of bundled files I'm getting issues with `Duplicate identifier: Promise`

Currently my only resolution is to download the bundled version and add manually.

**Comment by mikeybanez** [comment]
Thanks for this report!

@SML-MeSo let's take another look at our module publishing as a whole. @dcefram is informing me that our TypeScript tutorial is also outdated, and that he's getting multiple errors with a simple `yarn add` or `npm install`.

**Comment by cecilemuller** [comment]
There seems to be something incompatible with latest Typescript as well.

For example, [this example](https://github.com/wildpeaks/boilerplate-typescript-webpack-xsplit) compiles fine with TS 2.6.2, but 2.7.1 (the latest) cannot compile anymore.

Quickly looking at the dependencies of xjs, it sounds like it's written for TS 1.6 ?
If so, it's bound to break sooner than later unfortunately.

**Comment by mikeybanez** [comment]
Yes, unfortunately it's still at TS 1.6 (I believe that last time I checked this, it had to be at 1.6 for docgen reasons).

But internally we've already started discussing moving it to newer TS and fixing any remaining issues or limitations with the way it's being bundled and published.
